src/figuresClasses.ts

```diff
@@ -1,11 +1,77 @@
-export interface Figure {}
+export type Color = 'red' | 'green' | 'blue';
+export type Shape = 'triangle' | 'circle' | 'rectangle';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a >= b + c || b >= a + c || c >= a + b) {
+      throw new Error('put correct sides');
```

Comment:
The error message 'put correct sides' is not very descriptive. It would be better to explain what is wrong with the sides, for example, 'Triangle sides do not adhere to the triangle inequality theorem.'